tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -213,6 +222,12 @@ def test_get_type_match_failure(self):
       # `a_typecheck_fail_test` above expects `wrapper.inner` to be a `B`.
       self.scheduler.product_request(A, [Params(TypeCheckFailWrapper(A()))])
 
+  def test_unhashable_failure(self):
+    """Test that unhashable Get(...) params result in a structured error."""
+    expected_msg = """TypeError: unhashable type: 'list'"""
```

Comment:
I would like for this message to check more specifically which object has the unhashable type, but I can't see how to do that easily, so this is definitely specific enough for now.